hpe3parclient/client.py

```diff
@@ -494,8 +505,43 @@ def createVolume(self, name, cpgName, sizeMiB, optional=None):
 
         """
         info = {'name': name, 'cpg': cpgName, 'sizeMiB': sizeMiB}
+        # For merlin array there is no compression and tdvv fields
+        # removing tdvv and replacing compression with reduce field
+        if not optional and self.merlin_supported:
+            optional = {}
+            optional['tpvv'] = True
         if optional:
+            if self.merlin_supported:
+                if 'tdvv' in optional:
+                    optional.pop('tdvv')
+
+                if optional.get('tpvv') is True \
+                        and optional.get('compression') is True:
+                    optional['reduce'] = True
+
+                if not optional.get('tpvv') \
+                        and not optional.get('compression'):
+                    optional['tpvv'] = True
```

Comment:
I think the above two 'if' conditions are redundant - even if we remove them, there won't be any change in the behavior. Do let me know if you think otherwise.